src/openforms/formio/formatters/default.py

```diff
@@ -0,0 +1,122 @@
+from typing import Dict
+
+from openforms.plugins.plugin import AbstractBasePlugin
+
+from .registry import register
+
+
+@register("default")
+class DefaultFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        return str(value)
+
+
+@register("textfield")
+class TextFieldFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        return str(value)
+
+
+@register("email")
+class EmailFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        return str(value)
+
+
+@register("date")
+class DateFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("time")
+class TimeFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("phoneNumber")
+class PhoneNumberFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("file")
+class FileFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("textarea")
+class TextAreaFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("number")
+class NumberFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("password")
+class PasswordFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("checkbox")
+class CheckboxFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        raise NotImplementedError
+
+
+@register("selectboxes")
+class SelectBoxesFormatter(AbstractBasePlugin):
+
+    def format(self, component: Dict, value: str) -> str:
+        selected_labels = [
+            entry["label"]
+            for entry in component["values"]
+            if value.get(entry["value"])
+        ]
+        return ", ".join(selected_labels)
```

Comment:
@sergei-maertens just checking, is this sort of what you had in mind for the formatters part of https://github.com/open-formulieren/open-forms/issues/1068? My idea would be that you could use it in `Submission.get_printable_data` like this:

```python
formatter = register[info["type"]] if info["type"] in register else register["default"]
printable_data[label] = formatter.format(info, info["value"])
```

Reply:
yeah, that looks good!